diff -Nurp pot-old/src/config.h pot/src/config.h
--- pot-old/src/config.h	2016-06-29 13:28:09.619035108 -0400
+++ pot/src/config.h	2016-06-29 13:28:47.009034317 -0400
@@ -7,7 +7,7 @@
 #define POT_RESOLUTION	1		/* minute */
 #define POT_PORT	3333		/* potval listen this port */
 
-#define NAND_FLASH_BLOCKSIZE	(128 * 1024)	/* bytes, 128KB */
+#define NAND_FLASH_BLOCKSIZE	(124 * 1024)	/* bytes, 124KB */
 #define NAND_FLASH_PAGESIZE	(2 * 1024)	/* bytes, 2KB */
 #define POT_MTD_SIZE	(512 * 1024)	/* bytes, POT partition is 512KB(4 blocks) by default */
 #define FIRST_NTP_TIME_OFFSET	(2 * NAND_FLASH_BLOCKSIZE)
