diff -Nur linux-3.4.103/sourcecode/include/linux/atm.h linux-3.4.103-new/sourcecode/include/linux/atm.h
--- linux-3.4.103/sourcecode/include/linux/atm.h	2013-12-10 21:48:33.000000000 -0500
+++ linux-3.4.103-new/sourcecode/include/linux/atm.h	2015-04-11 11:15:48.704310491 -0400
@@ -139,6 +139,9 @@
 	int		min_pcr;	/* minimum PCR in cells per second */
 	int		max_cdv;	/* maximum CDV in microseconds */
 	int		max_sdu;	/* maximum SDU in bytes */
+	int     scr;		/* sustained rate in cells per second */
+	int     mbs;		/* maximum burst size (MBS) in cells */
+	int     cdv;		/* Cell delay varition */
         /* extra params for ABR */
         unsigned int 	icr;         	/* Initial Cell Rate (24-bit) */
         unsigned int	tbe;		/* Transient Buffer Exposure (24-bit) */ 
