diff -u -r libubox-2013-07-04-old/sh/jshn.sh libubox-2013-07-04/sh/jshn.sh
--- libubox-2013-07-04-old/sh/jshn.sh	2015-02-04 16:34:33.000000000 +0800
+++ libubox-2013-07-04/sh/jshn.sh	2015-02-04 16:25:32.000000000 +0800
@@ -80,7 +80,8 @@
 	if [ "${cur%%[0-9]*}" = "JSON_ARRAY" ]; then
 		_json_inc "SEQ_$cur" var
 	else
-		local name="${var//[^a-zA-Z0-9_]/_}"
+#		local name="${var//[^a-zA-Z0-9_]/_}"
+		local name="$(echo -n "$var" | tr -c 'a-zA-Z0-9_' '_')"
 		[[ "$name" == "$var" ]] || _json_export "NAME_${cur}_${name}" "$var"
 		var="$name"
 	fi
@@ -201,7 +202,8 @@
 	local __cur
 
 	_json_get_var __cur JSON_CUR
-	local __var="${JSON_PREFIX}TYPE_${__cur}_${2//[^a-zA-Z0-9_]/_}"
+#	local __var="${JSON_PREFIX}TYPE_${__cur}_${2//[^a-zA-Z0-9_]/_}"
+	local __var="${JSON_PREFIX}TYPE_${__cur}_$(echo -n "$2" | tr -c 'a-zA-Z0-9_' '_')"
 	eval "export -- \"$__dest=\${$__var}\"; [ -n \"\${$__var+x}\" ]"
 }
 
@@ -210,7 +212,8 @@
 	local __cur
 
 	_json_get_var __cur JSON_CUR
-	local __var="${JSON_PREFIX}${__cur}_${2//[^a-zA-Z0-9_]/_}"
+#	local __var="${JSON_PREFIX}${__cur}_${2//[^a-zA-Z0-9_]/_}"
+	local __var="${JSON_PREFIX}${__cur}_$(echo -n "$2" | tr -c 'a-zA-Z0-9_' '_')"
 	eval "export -- \"$__dest=\${$__var}\"; [ -n \"\${$__var+x}\" ]"
 }
 
