From f69bcd37c665680f058adfa77ec16ea7f9ce2564 Mon Sep 17 00:00:00 2001
From: figo.yan <figo.yan@dniserver.dnish.net>
Date: Fri, 8 Apr 2016 07:02:03 -0400
Subject: [PATCH] backup env PATH and restore it for net-cgi

Fix bug#59692

Signed-off-by: figo.yan <figo.yan@dniserver.dnish.net>
---
 src/uhttpd-cgi.c |   19 ++++++++++++++++++-
 1 files changed, 18 insertions(+), 1 deletions(-)

diff --git a/src/uhttpd-cgi.c b/src/uhttpd-cgi.c
index 57ef574..e9f7cd8 100755
--- a/src/uhttpd-cgi.c
+++ b/src/uhttpd-cgi.c
@@ -326,12 +326,29 @@ void uh_cgi_request(
 			     (pi->stat.st_mode & S_IXOTH)) || (ip != NULL) || ( binary != NULL )
 			) {
 				/* build environment */
+				char *path_p = NULL;
+				char *path_b = NULL;
+				path_p = getenv("PATH");
+				if ( path_p) {
+					path_b = strdup( path_p);
+					if( !path_b){
+						perror("strdup for PATH");
+						exit(errno);
+					}
+					/* No need to free path_b,
+					 * when this child process exits, it will be cleaned
+					 */
+				}
+
 				clearenv();
 
 				/* common information */
 				setenv("GATEWAY_INTERFACE", "CGI/1.1", 1);
 				setenv("SERVER_SOFTWARE", "uHTTPd", 1);
-				setenv("PATH", "/sbin:/usr/sbin:/bin:/usr/bin", 1);
+
+				if( path_b)
+					setenv("PATH", path_b, 1);
+
 
 #ifdef HAVE_TLS
 				/* https? */
-- 
1.7.4.1

