--- a/update_smb.c
+++ b/update_smb.c
@@ -80,6 +80,7 @@ struct share_info
 #endif
 #define USB_PATH_SIZE	4096
 #define USB_STORAGE_KEYWORD "/dev/sd"
+#define USB_STORAGE_KEYWORD_MAPPER "/dev/mapper/sd"
 
 extern char *config_get(char* name);
 extern void config_set(char *name, char *value);
@@ -98,7 +99,8 @@ static void reload_services(void)
 	struct statfs stat;
 	FILE *fp = setmntent("/proc/mounts", "r");
 	while ((ent = getmntent(fp))) {
-		if (strncmp(ent->mnt_fsname, USB_STORAGE_KEYWORD, strlen(USB_STORAGE_KEYWORD)))
+		if (strncmp(ent->mnt_fsname, USB_STORAGE_KEYWORD, strlen(USB_STORAGE_KEYWORD))
+			&& strncmp(ent->mnt_fsname, USB_STORAGE_KEYWORD_MAPPER, strlen(USB_STORAGE_KEYWORD_MAPPER)))
 			continue;
 
 		bzero(&stat, sizeof(struct statfs));
